code/datums/stars.dm

```diff
@@ -290,3 +286,88 @@
 		P.disp_dist = 0.25 * (4 ** (P.disp_level/planets.len))*1.3333 - 0.3333
 		P.disp_x = cos(P.disp_angle) * P.disp_dist
 		P.disp_y = sin(P.disp_angle) * P.disp_dist
+
+/datum/station_module
+	var/list/buy_keywords = list()		// Associated list of keywords and price modifiers. Price modifiers are multiplicative.
+	var/list/sell_keywords = list()		// As above, but for players selling to the station.
+	var/rarity = 50		// weight of a module to get picked.
+	var/datum/space_station/station
+	var/module_suffix = "Meta-Physical Object That Should Not Exist"		//suffix of a station. Flavortext!
+	var/name		//actual constructed name
+/datum/station_module/New(var/datum/space_station/S)
+	if(S)
+		station = S
```

Comment:
no need for an if here

Reply:
Needed for when station modules are created and then deleted for module weighting, unless you want those to dummy out and have them point to the starting planet

Reply:
station is null by default, is S is null then it's no problem if station = null

Reply:
Oh i know what i did wrong, herp i got an error when i deleted both of the ifs